plugin/src/test/java/io/jenkins/plugins/casc/vault/VaultTestUtil.java

```diff
@@ -0,0 +1,76 @@
+package io.jenkins.plugins.casc.vault;
+
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.Container;
+import org.testcontainers.vault.VaultContainer;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class VaultTestUtil {
+    private final static Logger LOGGER = Logger.getLogger(VaultTestUtil.class.getName());
+
+    public static final String VAULT_DOCKER_IMAGE = "vault:1.0.3";
+    public static final String VAULT_ROOT_TOKEN = "root-token";
+    public static final String VAULT_USER = "admin";
+    public static final String VAULT_URL = "http://localhost:8200";
+    public static final String VAULT_PW = "admin";
+    public static final String VAULT_POLCY= "io/jenkins/plugins/casc/vault/vaultTest_adminPolicy.hcl";
+    public static final String VAULT_PATH_V1 = "kv-v1/admin";
+    public static final String VAULT_PATH_V2 = "kv-v2/admin";
+
+    private static Container.ExecResult runCommand(VaultContainer container, final String... command) throws IOException, InterruptedException {
+        LOGGER.log(Level.FINE, String.join(" ", command));
+        final Container.ExecResult result = container.execInContainer(command);
+        final String out = result.getStdout();
+        final String err = result.getStderr();
+        if (out != null && !out.isEmpty()) {
+            LOGGER.log(Level.FINE, result.getStdout());
+        }
+        if (err != null && !err.isEmpty()) {
+            LOGGER.log(Level.WARNING, result.getStderr());
+        }
+        return result;
+    }
+
+    public static boolean isWindowsNode() {
+        return System.getProperty( "os.name" ).startsWith( "Windows" );
+    }
+
+    public static VaultContainer createVaultContainer() {
+        if (isWindowsNode()) return null;
+        return new VaultContainer<>(VaultTestUtil.VAULT_DOCKER_IMAGE)
+                .withVaultToken(VaultTestUtil.VAULT_ROOT_TOKEN)
+                .withClasspathResourceMapping(VAULT_POLCY, "/admin.hcl", BindMode.READ_ONLY)
+                .withVaultPort(8200);
+    }
+
+    public static void configureVaultContainer(VaultContainer container) {
+        if (isWindowsNode()) return;
+        try {
+            // TODO: proper life cycle or polling
+            // Give vault 5s to start
+            Thread.sleep(5000);
```

Comment:
Use the `waitingFor` feature of testcontainers on the `createVaultContainer` above
https://www.testcontainers.org/features/startup_and_waits/

https://github.com/BetterCloud/vault-java-driver/blob/cda433f56ab0670c79277ee15913f7628bd41878/src/test-integration/java/com/bettercloud/vault/util/VaultContainer.java#L55-L72

Reply:
Here is a simpler example:
https://github.com/testcontainers/testcontainers-java/blob/8cab5ce097ccb2fbfb6684ac2289496bddf105d2/modules/vault/src/test/java/org/testcontainers/vault/VaultContainerTest.java#L33

Reply:
Yes thats good. Thank you! 
Added it